### ISSUE TITLE
Compiling issue on Ubuntu 16.04

### ISSUE THREAD

**Issue by mrmookie** [furrtek/portapack-havoc#61]
I'm having difficulty building the firmware myself. I've attached a screenshot of the error.. 

![havoc](https://cloud.githubusercontent.com/assets/1279112/26036020/4286fb66-38d6-11e7-9813-0aa6670e948f.png)

I installed the package 'gcc-arm-none-eabi' I'm a bit confused if I have the right one installed. Thanks for any help.

**Comment by gh0stshell** [comment]
https://launchpad.net/gcc-arm-embedded/+download

**Comment by furrtek** [comment]
The link @gh0stshell gave should give you a version of gcc-arm-none-eabi which doesn't produce that error.

**Comment by mrmookie** [comment]
Thanks for the info, I've been about to get further but I seem to have a different error now.. 

![2017-05-18 16_44_36-root twitter_ _home_twitter_portapack-havoc_build](https://cloud.githubusercontent.com/assets/1279112/26208181/cb945f1e-3be9-11e7-9652-83d4d4490918.png)


Showing I'm using cmake version 3.8.1 and arm-none-eabi-gcc at 5.4.1.  Am I missing something obvious? Thanks for the help.

**Comment by furrtek** [comment]
Those 2 files not being spotted by cmake is still strange...
Do you have hackrf.img and terminator.img in firmware/baseband/ ?
They should be generated automatically in time, and anyways I included them in the repo so they should be there.

**Comment by mrmookie** [comment]
Yes, both hackrf.img and terminator.img appear in portapack-havoc/firmware/baseband/

**Comment by mrmookie** [comment]
I notice all the path's in /firmware/Makefile reference /home/furrtek... that path obviously doesn't exist on my machine. Could this be causing an issue?

**Comment by gh0stshell** [comment]
I had the same thought but after some help it worked even though I did not change the paths

You had the same issue as me at first, try the workaround

 copy hackrf.img and terminator.img from Sharebrained's repo and place them in the correct dir

**Comment by furrtek** [comment]
I shouldn't have committed the makefile with my home directory in it... I'll fix that shortly.

**Comment by mrmookie** [comment]
Ok so thinking about this... The missing files are in the repo /firmware/baseband/ but during build I think it's looking for the files (which are missing) in /build/firmware/baseband ? Does that help or make any sense? Once I placed the Sharebrained ones into the /build/firmware/baseband directory, the build completed!

**Comment by gh0stshell** [comment]
as part of the build did you do mkdir build?

**Comment by mrmookie** [comment]
Yes, just confirmed..

I have it working now, you need to copy the hackrf.img and terminator.imp files that are in /firmware/baseband to the build folder /build/firmware/baseband. You don't need to get it from the Sharebrained repo.

Yes, I followed the WIKI and ran mkdir build, cd build, cmake.., make firmware. Then copied the needed files to my build folder.
